lib/datasource.js

```diff
@@ -992,7 +1004,9 @@ DataSource.prototype.discoverModelProperties = function(modelName, options, cb)
   cb = cb || utils.createPromiseCallback();
 
   if (this.connector.discoverModelProperties) {
-    this.connector.discoverModelProperties(modelName, options, cb);
+    var nameMapper = options.nameMapper || this.mapName;
+    var name = nameMapper('table', modelName);
+    this.connector.discoverModelProperties(name, options, cb);
```

Comment:
This is part of the problem. Currently, the tests are failing because their expectation is unreasonable; they mock up a database-specific relationship between the Model and the table in postgres:
```
'name': 'CustomerTest',
        'options': {
          'idInjection': false,
          'postgresql': {
            'schema': 'public',
            'table': 'customer_test',
          },
        },
 // etc
```

As a result, this test is hiding the fact that the connector itself does not map the name of the Model (what the user would expect to see) to the database table.

The goal of this is ultimately to give them a default, let each connector override that default if necessary, and finally, give the user a way to arbitrarily map however they please if all else fails.